morgen: 3.5.9 -> 3.6.6

Jegp opened the pull request:
Upgrades Morgen from 3.5.9 to 3.6.6

This is a follow-up of #361802 

I tried making the PR as skimmable and easy-to-process as possible. I would love to hear how I can make the process easier on the maintainers. Thank you for the work you're doing here!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Jegp committed:
morgen: 3.5.9 -> 3.6.6

felbinger commented:
> I tried making the PR as skimmable and easy-to-process as possible. 

You failed :)
![image](https://github.com/user-attachments/assets/f2f525dd-f200-4f3c-91e8-f3b90a0de5d0)

You made the nix expression invalid.

Check the CI pipeline, after proposing the change, if it fails fix it!

If your confident, that your changes are ready to be merged contact the maintainers of the package (here: justanotherariel, see meta section of the file you edited...)

And again: Do not create a new pr for each version bump because a new patch has been released, simply update your pr.

Jegp commented:
> You failed :) ![image](https://private-user-images.githubusercontent.com/26925347/397992913-f2f525dd-f200-4f3c-91e8-f3b90a0de5d0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ4ODA2ODQsIm5iZiI6MTczNDg4MDM4NCwicGF0aCI6Ii8yNjkyNTM0Ny8zOTc5OTI5MTMtZjJmNTI1ZGQtZjIwMC00ZjNjLTkxZTgtZjNiOTBhMGRlNWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIyVDE1MTMwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhMmEzZDI3MzM4MTQ0MmY0MWM4YmI2MjE0MjE0ODMxMWM1ZWM3ZjUyMjVkZWViYTllOTVmMjczMmY4MzYxMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.bffPErs8PDeEMPgjnucPURFXZX2XdeSjR-fI_BQZwJ0)
> 
> You made the nix expression invalid.

Thank you! I fixed and tested the changes locally. I'll keep an eye on the checks.

> If your confident, that your changes are ready to be merged contact the maintainers of the package (here: justanotherariel, see meta section of the file you edited...)
> 

Thanks, that's helpful because it gives me agency. If all the tests pass, I'll reach out to them.

felbinger commented on pkgs/applications/office/morgen/default.nix:
```diff
@@ -3,14 +3,13 @@
-    hash = "sha256-ZKlj/QuQnrqQepsJY6KCROC2fXK/4Py5tmI/FVnRi9w=";
+    hash = "sha256-kq7g79uKzk9k0RQ2OqHPRJxiHCYceVlfCUQ300rVn5E=";
   };
-
```
I guess we want to keep this line ;)

felbinger commented:
Looks good so far, ofborg will take very long time to finish, but your change looks good. If you tested it locally and it all still works (which is the case usually for simple patch updates) you can ping him to check it too.
Due to the fact that morgen hasn't been merged to the new structure (pkgs/by-name), he won't be able to merge it though (He may have rights, I didn't check)... With the new structure package maintainers can merge their packages themself using a bot.

Jegp commented:
Line re-introduced :-)

> Looks good so far, ofborg will take very long time to finish, but your change looks good. If you tested it locally and it all still works (which is the case usually for simple patch updates) you can ping the maintainer to check it too. Due to the fact that morgen hasn't been moved to the new structure (pkgs/by-name) yet, he (at least as a simple package maintainer) won't be able to merge it though (He may have merge rights for the whole repo, I didn't check). With the new structure package maintainers can merge their packages themself using a bot.

Thanks for the information. I'm learning a lot! But if that's the case, shouldn't I just move the whole thing to `pkgs/by-name/mo/morgen`?

justanotherariel commented:
Yea, I planned doing that at some point but didn't get to it. Sorry for the late reply - things have been very busy latetly. Seems like something broke in the auto update script... that needs fixing too, then ryantm will take care of creating these PRs.
I'll test your PR, and then make my own PR moving morgen to the pkgs dir and fixing the update script - hopefully in the next couple of days.
Cheers

justanotherariel commented:
And you can (and should) test your chages in nixpkgs by checking out the repository/branch, and then executing `nix run .#morgen` in that directory, if you have nix experimental commands enabled. There's probably a way to do it with `nix-shell` too. If you need an update quickly, you can include your own git patches when importing nixpkgs in your flake - but that may be a bit too advanced.

Jegp commented:
No worries about a late reply. This is volunteer work. I must admit the process here is confusing to me, but I am just trying to be helpful.

I did test it locally. That's the version I'm running now.

I believe the trailing whitespace was fixed after I addressed the comment from @felbinger

felbinger commented on pkgs/applications/office/morgen/default.nix:
```diff
@@ -3,14 +3,14 @@
+    hash = "sha256-kq7g79uKzk9k0RQ2OqHPRJxiHCYceVlfCUQ300rVn5E=";
   };
-
+  
```
You added two spaces in an empty line... Remove them:
```suggestion

```

Jegp replied:
Done

Jegp commented:
Branch has been updated with latest version of Morgen, 3.6.6

felbinger commented:
> Branch has been updated with latest version of Morgen, 3.6.6

Please rebase, merge commits aren't allowed!
![image](https://github.com/user-attachments/assets/51d63f57-5176-477c-aec4-9e0f90ce9c6a)

felbinger commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 367347`

Logs: https://github.com/felbinger/nixpkgs-review-gha/actions/runs/13330340733


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>morgen</li>
  </ul>
</details>

Jegp commented:
> > Branch has been updated with latest version of Morgen, 3.6.6
> 
> Please rebase, merge commits aren't allowed! 

Turns out that rebasing was a pretty poor strategy. It added a ton of partial or unresolved commits. Which surprises me, because I would assume nixpkgs commit history shouldn't contain that kind of discrepancies. Do you know why that's the case?

Anyway, I reset to `nixpkgs` master and cherry-picked the commit for Morgen. So it should be up-to-date now.
